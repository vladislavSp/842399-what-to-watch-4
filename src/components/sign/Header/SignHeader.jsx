import React from 'react';
import PropTypes from 'prop-types';
import {AuthorizationStatus} from '../../../reducer/user/user';
import {getAuthorizationStatus, getUser} from '../../../reducer/user/selectors';
import {Link} from 'react-router-dom';
import {connect} from 'react-redux';
import {AVATAR_BASE_PATH} from '../../../const/const';

export const SignHeader = ({status, user}) => {
  const getHeaderByStatus = (authStatus) => {
    switch (authStatus) {
      case AuthorizationStatus.AUTH:
        return <Link to={`/mylist`}>
          <div className="user-block__avatar">
            <img src={`${AVATAR_BASE_PATH}${user.avatar}`} alt="User avatar" width="63" height="63" />
          </div>
        </Link>;
      default:
        return <Link to={`/login`} href="" className="user-block__link">Sign in</Link>;
    }
  };

  return (
    <div className="user-block">
      { getHeaderByStatus(status) }
    </div>
  );
};

const mapStateToProps = (state) => ({
  user: getUser(state),
  status: getAuthorizationStatus(state),
});

SignHeader.propTypes = {
  status: PropTypes.string.isRequired,
  user: PropTypes.object,
};

export default connect(mapStateToProps)(SignHeader);
