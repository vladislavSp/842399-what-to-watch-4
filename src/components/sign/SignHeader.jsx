import React from 'react';
import PropTypes from 'prop-types';
import {AuthorizationStatus} from '../../reducer/user/user';

const SignHeader = ({status, onSignInClick}) => {
  const getHeaderByStatus = (authStatus) => {
    switch (authStatus) {
      case AuthorizationStatus.AUTH:
        return <div className="user-block">
          <div className="user-block__avatar">
            <img src="img/avatar.jpg" alt="User avatar" width="63" height="63" />
          </div>
        </div>;
      default:
        return <div className="user-block">
          <a
            onClick={onSignInClick}
            href="#"
            className="user-block__link">Sign in</a>
        </div>;
    }
  };

  return getHeaderByStatus(status);
};

SignHeader.propTypes = {
  status: PropTypes.string.isRequired,
  onSignInClick: PropTypes.func.isRequired,
};

export default SignHeader;
