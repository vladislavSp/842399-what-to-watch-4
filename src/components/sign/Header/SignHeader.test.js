import React from 'react';
import renderer from 'react-test-renderer';
import SignHeader from './SignHeader.jsx';

const status = `AUTH`;

describe(`render Sign Header`, () => {
  it(`Sign Header render correctly`, () => {
    const tree = renderer.create((
      <SignHeader
        status={status}
        onSignInClick={() => {}}
      />
    )).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
