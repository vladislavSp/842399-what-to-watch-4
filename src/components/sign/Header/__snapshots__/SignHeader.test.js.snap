// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Sign Header Sign Header render correctly 1`] = `
<div
  className="user-block"
>
  <div
    className="user-block__avatar"
  >
    <img
      alt="User avatar"
      height="63"
      src="img/avatar.jpg"
      width="63"
    />
  </div>
</div>
`;
